bootstrap/golang/prep-new-repo.sh

```diff
@@ -0,0 +1,112 @@
+set -euo pipefail
+
+# =============================================================================
+#                                    Constants
+# =============================================================================
+LAMBDA_IMPL_DIRNAME="kurtosis-lambda"
+SCRIPTS_DIRNAME="scripts"
+
+GO_MOD_FILENAME="go.mod"
+GO_MOD_MODULE_KEYWORD="module "  # The key we'll look for when replacing the module name in go.mod
+
+BUILD_SCRIPT_FILENAME="build.sh"
+BUILD_SCRIPT_KEYWORD="IMAGE_NAME"
```

Comment:
Let's make this explicit to what the var actually is - `BUILD_SCRIPT_IMAGE_NAME_VAR_NAME`